### ISSUE TITLE
The model starts training from scratch during stage 2, 3

### ISSUE THREAD

**Issue by alsabahi** [grammarly/gector#42]
I have replicated your xlnet based result during stage 1 but it seems that the model starts training from scratch during stage 2, even though the pretrain arg is set to the best model from stage 1. Am I missing something. I am new to allennlp library. Here is the command I used for stage 2:
`python train.py --train_set "./data/wi.train.tagged" --dev_set "./data/wi.dev.tagged" --model_dir "./modelxlnet_stage2/" --vocab_path=data/output_vocabulary --skip_correct=1 --tp_prob 1 --tn_prob 0 --special_tokens_fix=0 --cold_steps_count=2 --n_epoch 20 --transformer_model=xlnet  --updates_per_epoch=0 --batch_size=64 --accumulation_size=2 --pretrain_folder ./modelxlnet_stage1/ --pretrain best`

**Comment by komelianchuk** [comment]
Why do you think so?
The error distribution is a bit different in stage2, so it might start from higher loss.

**Comment by alsabahi** [comment]
Lets say that the best checkpoint from stage 1 is the 6th one. I suppose that in the next stage, the next checkpoint should be checkpoint 7 not checkpoint 1. Another thing, after few epochs of training, the F0.5 should be more than it was during stage 1 best checkpoint . Correct me if I am mistaken.

**Comment by alsabahi** [comment]
Here is BEA dev F0.5 after 20 epochs:
![image](https://user-images.githubusercontent.com/33044055/89022130-32a54300-d32a-11ea-8152-949cc610186f.png)
and here is the training metrics:
```
{
  "best_epoch": 19,
  "peak_cpu_memory_MB": 2876.124,
  "peak_gpu_0_memory_MB": 9018,
  "peak_gpu_1_memory_MB": 11,
  "peak_gpu_2_memory_MB": 11,
  "peak_gpu_3_memory_MB": 11,
  "peak_gpu_4_memory_MB": 11,
  "peak_gpu_5_memory_MB": 11,
  "peak_gpu_6_memory_MB": 11,
  "peak_gpu_7_memory_MB": 3474,
  "training_duration": "0:32:31.716916",
  "training_start_epoch": 0,
  "training_epochs": 19,
  "epoch": 19,
  "training_accuracy": 0.8567961645898132,
  "training_loss": 0.8852979070341527,
  "training_cpu_memory_MB": 2876.124,
  "training_gpu_0_memory_MB": 6430,
  "training_gpu_1_memory_MB": 11,
  "training_gpu_2_memory_MB": 11,
  "training_gpu_3_memory_MB": 11,
  "training_gpu_4_memory_MB": 11,
  "training_gpu_5_memory_MB": 11,
  "training_gpu_6_memory_MB": 11,
  "training_gpu_7_memory_MB": 3410,
  "validation_accuracy": 0.8880430691205896,
  "validation_loss": 0.6560100949355189,
  "best_validation_accuracy": 0.8880430691205896,
  "best_validation_loss": 0.6560100949355189
}
```

**Comment by alsabahi** [comment]
I am supposing the loss function to start from a better position:
```
{
  "best_epoch": 0,
  "peak_cpu_memory_MB": 2876.124,
  "peak_gpu_0_memory_MB": 6988,
  "peak_gpu_1_memory_MB": 11,
  "peak_gpu_2_memory_MB": 11,
  "peak_gpu_3_memory_MB": 11,
  "peak_gpu_4_memory_MB": 11,
  "peak_gpu_5_memory_MB": 11,
  "peak_gpu_6_memory_MB": 11,
  "peak_gpu_7_memory_MB": 1238,
  "training_duration": "0:00:57.773522",
  "training_start_epoch": 0,
  "training_epochs": 0,
  "epoch": 0,
  "training_accuracy": 0.7865364423400757,
  "training_loss": 1.8930613819112632,
  "training_cpu_memory_MB": 2876.124,
  "training_gpu_0_memory_MB": 6988,
  "training_gpu_1_memory_MB": 11,
  "training_gpu_2_memory_MB": 11,
  "training_gpu_3_memory_MB": 11,
  "training_gpu_4_memory_MB": 11,
  "training_gpu_5_memory_MB": 11,
  "training_gpu_6_memory_MB": 11,
  "training_gpu_7_memory_MB": 1238,
  "validation_accuracy": 0.7943741704142481,
  "validation_loss": 1.6749383416877786,
  "best_validation_accuracy": 0.7943741704142481,
  "best_validation_loss": 1.6749383416877786
}
```

**Comment by komelianchuk** [comment]
I see. I can suggest the following:
1. Try to debug the loading of the model and ensure that the weights of the pre-trained model are correctly loaded.
2. Check your training data for stage2. For example if you occasionally swap source/target for some dataset it might explain this weird behavior.

**Comment by alsabahi** [comment]
@komelianchuk  thanks for the suggestion. The issue has been solved by the reprocessing the training data for stage 2.
